elephant/spike_train_correlation.py

```diff
@@ -117,12 +124,17 @@ def corrcoef(binned_sts, binary=False):
         :math:`C[i,j]=C[j,i]` is the Pearson's correlation coefficient between
         binned_sts[i] and binned_sts[j]. If binned_sts contains only one
         SpikeTrain, C=1.0.
+    mask : ndarray
```

Comment:
Reading through the code, we now understand the behavior that either the mask or the correlation matrix is returned depending on  `with_nans`. This behavior is confusing, since it this means that a function called "corrcoeff" outputs a list of non-empty spike trains for this parameter setting.

We suggest the following: 
* Remove the `with_nans` parameter from the function
* Keep the functionality that empty spike trains always become a NaN and that a warning is issued. 
* We add to the BinnedSpiketrain class functions of the type `non_empty_mask()`, `empty_mask()`, `non_empty_ids()`... similar to the functionality that had been implemented here.
* We use these functions in re-coding the correlation and covariance functions.

Reply:
If we keep the old behaviour then the warning needs to be more meaningful. This PR was induced by issue https://github.com/NeuralEnsemble/elephant/issues/169. So we can stay in terms with the issue.

Reply:
Btw. a `BinnedSpikeTrain` is in its core a sparse matrix, which already provides already masking operations. For example one could call the function `to_sparse_array().nonzero()` which gives back non zero rows and columns indices.